subsys/net/openthread/rpc/server/ot_rpc_dns_client.c

```diff
@@ -0,0 +1,689 @@
+			 (otError error, const otDnsBrowseResponse *response, void *context),
+			 (error, response, context));
+
+static void service_response_callback(otError error, const otDnsServiceResponse *response,
+				     void *context, uint32_t callback_slot)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, ((2 * sizeof(uint32_t)) + 2) +
+					   ((2 * sizeof(uintptr_t)) + 2));
+
+	nrf_rpc_encode_uint(&ctx, (uint32_t) error);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)response);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)context);
+	nrf_rpc_encode_uint(&ctx, callback_slot);
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_DNS_SERVICE_RESPONSE_CB, &ctx,
+				ot_rpc_decode_void, NULL);
+}
+
+NRF_RPC_CBKPROXY_HANDLER(service_response_callback_encoder, service_response_callback,
+			 (otError error, const otDnsServiceResponse *response, void *context),
+			 (error, response, context));
+
+static void resolve_address(enum ot_rpc_cmd_server cmd, const struct nrf_rpc_group *group,
+			    struct nrf_rpc_cbor_ctx *ctx)
+{
+	otDnsQueryConfig config;
+	otDnsAddressCallback cb;
+	otError error = OT_ERROR_FAILED;
+	char hostname[OT_DNS_MAX_NAME_SIZE];
+	void *cb_ctx;
+	bool decoded_config;
+
+	nrf_rpc_decode_str(ctx, hostname, sizeof(hostname));
+
+	cb = (otDnsAddressCallback)nrf_rpc_decode_callbackd(ctx, address_response_callback_encoder);
+	cb_ctx = (void *)nrf_rpc_decode_uint(ctx);
+
+	decoded_config = ot_rpc_decode_dns_query_config(ctx, &config);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		ot_rpc_report_cmd_decoding_error(cmd);
+		return;
+	}
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+
+	switch (cmd) {
+	case OT_RPC_CMD_DNS_CLIENT_RESOLVE_ADDRESS:
+		error = otDnsClientResolveAddress(openthread_get_default_instance(), hostname, cb,
+						  cb_ctx, (decoded_config ? &config : NULL));
+		break;
+	case OT_RPC_CMD_DNS_CLIENT_RESOLVE_IP4_ADDRESS:
+		error = otDnsClientResolveIp4Address(openthread_get_default_instance(), hostname,
+						     cb, cb_ctx, (decoded_config ? &config : NULL));
+		break;
+	default:
+		error = OT_ERROR_INVALID_COMMAND;
+		break;
+	}
+
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+static void ot_rpc_dns_client_resolve_address(const struct nrf_rpc_group *group,
+					      struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	ARG_UNUSED(handler_data);
+
+	resolve_address(OT_RPC_CMD_DNS_CLIENT_RESOLVE_ADDRESS, group, ctx);
+}
+
+static void ot_rpc_dns_client_resolve_ip4_address(const struct nrf_rpc_group *group,
+						  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	ARG_UNUSED(handler_data);
+
+	resolve_address(OT_RPC_CMD_DNS_CLIENT_RESOLVE_IP4_ADDRESS, group, ctx);
+}
+
+static void ot_rpc_dns_client_browse(const struct nrf_rpc_group *group,
+				     struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	char service_name[OT_DNS_MAX_LABEL_SIZE];
```

Comment:
`OT_DNS_MAX_LABEL_SIZE` may be too small. The service name can be a full DNS name, not just a single component.

Reply:
Good point, fixed